src/apps/lwaftr/icmp.lua

```diff
@@ -93,6 +95,22 @@ function new_icmpv4_packet(from_eth, to_eth, from_ip, to_ip, initial_pkt, l2_siz
    return new_pkt
 end
 
+function is_icmpv4(pkt)
+   return is_ipv4(pkt)
+      and pkt.data[constants.ethernet_header_size + constants.o_ipv4_proto] == constants.proto_icmp
```

Comment:
please cache these constants locally, i saw important speedups when doing so in the lwaftr.  e.g. local ethernet_header_size = constants.ethernet_header_size.  Ideally you would use a strategy like the one in lwaftr.lua though; wdyt?